pandas/tests/series/test_apply.py

```diff
@@ -576,3 +576,13 @@ def f(x):
         result = s.map(f)
         exp = pd.Series(['Asia/Tokyo'] * 25, name='XX')
         tm.assert_series_equal(result, exp)
+
+    @pytest.mark.parametrize("vals,mapping,exp", [
+        (list('abc'), {np.nan: 'not NaN'}, ['not NaN']),
+        (list('abc'), {'string': 'another string'}, ['another string']),
+        (list(range(3)), {42: 'the answer'}, ['the answer'])])
+    def test_map_missing_mixed(self, vals, mapping, exp):
+        s = pd.Series(vals + [list(mapping.keys())[0]])
+        result = s.map(mapping)
+
+        tm.assert_series_equal(result[-1:].reset_index(drop=True), pd.Series(exp))
```

Comment:
Think through the exp values you are passing in. They should obviously match the shape of your input but replace with NA values where appropriate.

For your first example, if you did `list('abc')` as your val `{'a': 'foo'}` as your mapping then your exp would be `['foo', np.nan, np.nan, np.nan]`.

I'm not sure what you are trying to do with `result[-1:].reset_index(drop=True)` but that's getting way too complicated. If you follow all of the above steps you can just do `tm.assert_series_equal(result, pd.Series(exp))`